src/xunit.performance.api/Profilers/Etw/AddressSpace.cs

```diff
@@ -0,0 +1,102 @@
+    /// </summary>
+    internal sealed class AddressSpace : IInterval<ulong>
+    {
+        /// <summary>
+        /// Initializes a new instance of the AddressSpace class.
+        /// </summary>
+        /// <param name="address">Interval's start address.</param>
+        /// <param name="size">Interval's size.</param>
+        public AddressSpace(ulong address, uint size)
+        {
+            Start = address;
+            Size = size;
+            End = Start + Size;
+        }
+
+        /// <summary>
+        /// The size of the address space.
+        /// </summary>
+        public uint Size { get; }
+
+        /// <summary>
+        /// The address where the module was loaded.
+        /// </summary>
+        public ulong Start { get; }
+
+        /// <summary>
+        /// Start + Size
+        /// </summary>
+        public ulong End { get; }
+
+        /// <summary>
+        /// Determines if the specified address falls within this address space.
+        /// </summary>
+        /// <param name="address">Address to test against this address space.</param>
+        /// <returns>True if the address is within this interval, false otherwise.</returns>
+        public bool IsInInterval(ulong address)
+        {
+            return (Start <= address) && (address < End);
+        }
+
+        /// <summary>
+        /// Determines whether the specified object is equals to this object.
+        /// </summary>
+        /// <param name="obj">The object to compare with this object.</param>
+        /// <returns>True if the specified object is equal to the current object; otherwise, false.</returns>
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
+                return false;
+
+            var addressSpace = obj as AddressSpace;
+            if (addressSpace == null)
+                return false;
+
+            return Start == addressSpace.Start && Size == addressSpace.Size;
+        }
+
+        /// <summary>
+        /// Serves as the default hash function.
+        /// </summary>
+        /// <returns>A hash code for the current object.</returns>
+        public override int GetHashCode()
+        {
+            return Start.GetHashCode() ^ Size.GetHashCode();
+        }
+
+        /// <summary>
+        /// Determines whether two specified AddressSpace have the same value.
+        /// </summary>
+        /// <param name="lhs">The first AddressSpace to compare, or null.</param>
+        /// <param name="rhs">The second AddressSpace to compare, or null.</param>
+        /// <returns>True if its two operands refer to the same object or if the values of its operands are equal; otherwise, false.</returns>
+        public static bool operator ==(AddressSpace lhs, AddressSpace rhs)
+        {
+            if (ReferenceEquals(lhs, rhs))
+                return true;
+
+            if ((object)lhs == null || (object)rhs == null)
```

Comment:
Any reason not to call AddressSpace.Equals here?  Then you don't have to implement the equality comparison twice (especially if new conditions get added over time).